### DISCUSSION TITLE
Add a Config for multipart.NewWriter

### DISCUSSION THREAD

**Discussion by earthboundkid** [earthboundkid/requests#36] [Ideas] [answered]
API could look like `.Config(requests.BodyMultipart(func (w *multipart.Writer) error { /* ... */ }))`. It could handle setting the content type and closing the writer junk.

**Comment by Phathdt** [comment]
do we have example for body multypart like this postman ?

<img width="686" alt="image" src="https://user-images.githubusercontent.com/22195771/155958103-c7b202ad-6fc4-4f0b-92e6-fbc4d3a2e799.png">
 @carlmjohnson

**Reply by earthboundkid** [reply]
If I understand correctly, your body is form data, not necessarily multipart, so you can just do

```go
chatpng, _ := os.ReadFile("chat.png")

err := requests.
  URL(requrl).
  BodyForm(url.Values{
    "image": chatpng,
  }).
  Fetch(ctx)
```

If that won't work, someone will need to actually write a multipart mime encoder config. I looked into it once but have not needed it personally, so it's not written.

**Comment by Crocmagnon** [comment]
I just had the need to send multipart data to a picky server. Here's what I came up with, if others are interested:
```go
boundary := randomBoundary()

err := requests.URL(url).
	Method(http.MethodPost).
	Header("Content-Type", "multipart/form-data; boundary="+boundary).
	BodyWriter(func(w io.Writer) error {
		multi := multipart.NewWriter(w)

		if err := multi.SetBoundary(boundary); err != nil {
			return fmt.Errorf("setting boundary: %w", err)
		}

		writer, err := multi.CreateFormFile("file", filename)
		if err != nil {
			return fmt.Errorf("creating form file: %w", err)
		}

		if _, err := io.Copy(writer, reader); err != nil {
			return fmt.Errorf("copying file: %w", err)
		}

		if err := multi.Close(); err != nil {
			return fmt.Errorf("closing multipart writer: %w", err)
		}

		return nil
	}).
	Fetch(ctx)
```

**Reply by earthboundkid** [reply]
I opened a PR: https://github.com/earthboundkid/requests/pull/119/files

**Comment by earthboundkid** [comment, marked as answer]
https://github.com/earthboundkid/requests/pull/119/files was merged.
